src/obu.c

```diff
@@ -439,6 +439,11 @@ avifBool avifSequenceHeaderParse(avifSequenceHeader * header, const avifROData *
             return AVIF_FALSE;
 
         if (obu_type == 1) { // Sequence Header
+            if (obu_size == 0) {
+                // This has to be checked for explicitly because avifBitsInit
+                // does not validate whether there is at least one byte or not.
+                return AVIF_FALSE;
+            }
```

Comment:
I suggest including <assert.h> and adding `assert(size != 0)` to the beginning of `avifBitsInit`.

Alternatively, we can change line 69 from

```
    bits->eof = 0;
```

to

```
    bits->eof = (size == 0);
```

Reply:
Good point. I have updated the code to set eof to size == 0 since that is a better fix.